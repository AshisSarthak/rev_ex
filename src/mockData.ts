export const demoPacketData = {
  balance: 100,
  transactions: [
    {
      id: 1,
      timestamp: 1000230,
      balance: 50,
    },
    {
      id: 2,
      timestamp: 1000230,
      balance: 80,
    },
    {
      id: 3,
      timestamp: 1000230,
      balance: 90,
    },
  ],
};

export enum CURRENCY_SYMBOL {
  USD = "$",
  EUR = "€",
  GBP = "£",
}

export enum CURRENCYS {
  USD = "USD",
  EUR = "EUR",
  GBP = "GBP",
}
