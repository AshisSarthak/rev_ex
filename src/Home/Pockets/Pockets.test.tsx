import React from "react";
import { shallow } from "enzyme";
import { CURRENCYS, demoPacketData, CURRENCY_SYMBOL } from "../../mockData";
import Pocket, { getCurrencySymbol } from "./Pockets";

describe("Pockets Element", () => {
  it("gets the respective symbol", () => {
    expect(getCurrencySymbol(CURRENCYS.EUR)).toBe(CURRENCY_SYMBOL.EUR);
    expect(getCurrencySymbol(CURRENCYS.USD)).toBe(CURRENCY_SYMBOL.USD);
    expect(getCurrencySymbol(CURRENCYS.GBP)).toBe(CURRENCY_SYMBOL.GBP);
  });

  it("Loads the default Dropdown", () => {
    const wrapper = shallow(
      <Pocket pocket={undefined} currency={CURRENCYS.EUR} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("Loads the Pockets component with full data", () => {
    const wrapper = shallow(
      <Pocket pocket={demoPacketData} currency={CURRENCYS.EUR} />
    );

    const symbol = wrapper.find(".currency-symbol").at(0);
    const balance = wrapper.find(".currency-balance").at(0);

    expect(symbol.children().at(0).text()).toBe(
      getCurrencySymbol(CURRENCYS.EUR)
    );
    expect(balance.children().at(0).text()).toBe(
      demoPacketData.balance.toFixed(2)
    );

    const tableRow = wrapper.find(".exchange-entries");
    const individualRow = tableRow.at(1);
    expect(individualRow.children().at(0).text()).toBe(
      demoPacketData.transactions[0].timestamp + ""
    );
    expect(individualRow.children().at(1).text()).toBe(
      `${demoPacketData.transactions[0].balance}${getCurrencySymbol(
        CURRENCYS.EUR
      )}`
    );
    expect(tableRow).toHaveLength(demoPacketData.transactions.length + 1);
  });

  it("Loads the Pockets component with blank transaction Data data", () => {
    const newProps = { ...demoPacketData };
    newProps.transactions = [];
    const wrapper = shallow(
      <Pocket pocket={newProps} currency={CURRENCYS.EUR} />
    );

    const symbol = wrapper.find(".currency-symbol").at(0);
    const balance = wrapper.find(".currency-balance").at(0);
    expect(symbol.children().at(0).text()).toBe(
      getCurrencySymbol(CURRENCYS.EUR)
    );
    expect(balance.children().at(0).text()).toBe(
      demoPacketData.balance.toFixed(2)
    );
    const tableRow = wrapper.find(".exchange-entries");
    expect(tableRow).toHaveLength(newProps.transactions.length + 1);
  });
});
