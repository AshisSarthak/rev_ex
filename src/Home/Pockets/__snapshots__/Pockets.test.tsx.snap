// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pockets Element Loads the default Dropdown 1`] = `
<section
  className="pocket-container"
>
  <div
    className="pocket-currency"
  >
    <div
      className="currency-balance"
    >
      <div
        className="currency-symbol"
      >
        â‚¬
      </div>
    </div>
  </div>
  <div
    className="exchange-table"
  />
</section>
`;
